src/test/java/org/cactoos/io/TempFileTest.java

```diff
@@ -0,0 +1,114 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.io;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.cactoos.func.MatcherOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Unit tests for {@link TempFile}.
+ *
+ * @author George Aristy (george.aristy@gmail.com)
+ * @version $Id$
+ * @since 1.0
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+public final class TempFileTest {
+
+    @Test
+    public void tmpFileIsCreated() throws Exception {
+        try (final TempFile file = new TempFile()) {
+            MatcherAssert.assertThat(
+                "Cannot create a temp file",
+                Files.exists(file.value()),
+                Matchers.is(true)
+            );
+        }
+    }
+
+    @Test
+    public void createFileInCustomPath() throws Exception {
+        final Path custom = Paths.get(System.getProperty("user.home"));
+        try (final TempFile file = new TempFile(() -> custom, "", "")) {
+            MatcherAssert.assertThat(
+                "Cannot create a temp file at a custom path",
+                file,
+                Matchers.allOf(
+                    new MatcherOf<>(
+                        tmp -> {
+                            return Files.exists(tmp.value());
+                        }
+                    ),
+                    new MatcherOf<>(
+                        tmp -> {
+                            return tmp.value().getParent().equals(custom);
+                        }
+                    )
+                )
+            );
+        }
+    }
+
+    @Test
+    public void closedFileCeasesToExist() throws Exception {
+        final TempFile file = new TempFile();
+        file.close();
+        MatcherAssert.assertThat(
+            "Cannot delete file on close",
+            Files.exists(file.value()),
+            Matchers.is(false)
+        );
+    }
+
+    @Test
+    public void prefix() throws Exception {
+        final String prefix =
+            String.format("randomPrefix%s", System.currentTimeMillis());
+        try (final TempFile file = new TempFile(prefix, "")) {
+            MatcherAssert.assertThat(
+                "File not created with the given prefix",
+                file.value().getFileName().toString(),
+                Matchers.startsWith(prefix)
+            );
+        }
+    }
+
+    @Test
+    public void suffix() throws Exception {
+        final String suffix =
```

Comment:
@llorllale Does qulice validation pass with this indentation? I usually indent this code like this:

```java
final String suffix = String.format(
     "randomSuffix%s", System.currentTimeMillis()
);
```

But, if qulice passes, it's ok to leave it like that.

Reply:
qulice passes, but will change anyway because I too prefer this format